### DISCUSSION TITLE
Teleport of Reduced Coordinate Articulation

### DISCUSSION THREAD

**Discussion by mlp1802** [NVIDIA-Omniverse/PhysX#118] [Q&A]
So I've finally made the conversion from Max Coordinate to Reduced Coordinate articulation (thanks Philipp!). 
Now there's a new problem: in my game I frequently need to move and rotate the ragdolls. 
I've tried two methods.
1) Calling set global pose on each link. That produces a "NpArticulationLink::setGlobalPose teleport isn't allowed in the reduced coordinate system" error

2) Calling the "teleportRootLink" method. That works with regards to translation, the ragdoll can move around..but it **only** rotates the rootlink..the rest of the ragdoll stays in the same place. After a couple of frames with that the simulation explodes.

So my question is, what is the correct way to "teleport" a reduced coord articulation, both with respect to translation and rotation?

**Comment by mlp1802** [comment]
There's really no solution to this? I'm finding it hard to use articulations in any serious way without being able to position them. Been vacum cleaning the docs and the internet for a solution. Even asked chatgtp who said I could attach a rigid body to each link using a fixed joint, then move that body around which would make the articulation link move with it. Pretty creative but I'm not sure that is a very effective way.

**Comment by preist-nvidia** [comment]
Hi @mlp1802 - are you using this? https://github.com/NVIDIA-Omniverse/PhysX/blob/release/104.2/physx/include/PxArticulationReducedCoordinate.h#L1159

This will rotate the root link and all descendant links for you. It is correct that you cannot set the poses of the descendant links directly, because those are set implicitly by the root pose + the reduced joint coordinates.

**Reply by mlp1802** [reply]
Thanks for the reply! I'm using the "teleportRootLink" function in 4.1..there are no setGlobalPose in 4.1.

Does the teleportRootLink behave differently?

You can see its current behavior here (look at the root bone, between body and legs, the black small cube, it's rotated 45 degrees here around the z axis, and the rest of the body doesn't follow).

Maybe I should just upgrade to 5? 
![Screenshot from 2023-03-20 23-49-06](https://user-images.githubusercontent.com/4717892/226482697-0ac5fa44-4bf5-4ff5-a3d1-c06ad325279a.png)

**Reply by preist-nvidia** [reply]
Hmm, can be that you are hitting a bug in teleportRootLink then, I don't know if we had issues with that though.

Can you try setting the root pose using the articulation cache interface and see if it makes a difference? https://github.com/NVIDIAGameWorks/PhysX/blob/4.1/physx/include/PxArticulationReducedCoordinate.h#L88

Otherwise yes, I would consider upgrading to 5.1. We can provide better support with the current version.

**Reply by mlp1802** [reply]
I will try to upgrade then, thanks
